virttest/utils_test/qemu.py

```diff
@@ -1470,6 +1472,99 @@ def __init__(self, test, params, env):
         self.params = params
         self.sessions = {}
 
+    @error_context.context_aware
```

Comment:
Make this method as a classmethod and instead vm as session, so that other case can call this function without class instance. it will decrease overhead.
Other parts looks good to me.